components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/definitions/APIDefinitionFromSwagger12.java

```diff
@@ -0,0 +1,418 @@
+
+                    resourcePathJson.put("apiVersion", version);
+                    resourcePathJson.put("resourcePath", resourceName);
+                    resourceNameJSONs.put(resourceName, resourcePathJson);
+
+                    resourcePaths = new ArrayList<String>();
+                    resourcePaths.add(path);
+
+                    String httpVerbsStrng = template.getMethodsAsString();
+                    String[] httpVerbs = httpVerbsStrng.split(" ");
+                    for (String httpVerb : httpVerbs) {
+                        final JSONObject operationJson = (JSONObject) parser.parse(operationJsonTemplate);
+                        operationJson.put("method", httpVerb);
+                        operationJson.put("auth_type", template.getAuthType());
+                        operationJson.put("throttling_tier", template.getThrottlingTier());
+
+                        if(resourcePathJSONs.get(path) != null) {
+                            resourcePathJSONs.get(path).add(operationJson);
+
+                        } else {
+                            resourcePathJSONs.put(path, new ArrayList<JSONObject>() {{
+                                add(operationJson);
+                            }});
+                        }
+                    }
+                    resourceNamepaths.put(resourceName, resourcePaths);
+                }
+            }
+
+            for (Map.Entry<String, List<String>> entry : resourceNamepaths.entrySet()) {
+                String resourcePath = entry.getKey();
+                JSONObject jsonOb = resourceNameJSONs.get(resourcePath);
+                List<String> pathItems = entry.getValue();
+                for (String pathItem : pathItems) {
+                    JSONObject pathJson = (JSONObject) parser.parse(pathJsonTemplate);
+                    pathJson.put("path", pathItem);
+                    List<JSONObject> methodJsons = resourcePathJSONs.get(pathItem);
+                    for (JSONObject methodJson : methodJsons) {
+                        JSONArray operations = (JSONArray) pathJson.get("operations");
+                        operations.add(methodJson);
+                    }
+                    JSONArray apis1 = (JSONArray) jsonOb.get("apis");
+                    apis1.add(pathJson);
+                }
+            }
+
+            mainAPIJson.put("apiVersion", version);
+            ((JSONObject)mainAPIJson.get("info")).put("description", description);
+            for (Map.Entry<String, List<String>> entry : resourceNamepaths.entrySet()) {
+                String resourcePath = entry.getKey();
+                JSONObject jsonOb = resourceNameJSONs.get(resourcePath);
+                JSONArray apis1 = (JSONArray) mainAPIJson.get("apis");
+                JSONObject pathjob = new JSONObject();
+                pathjob.put("path",resourcePath);
+                pathjob.put("description","");
+                pathjob.put("file",jsonOb);
+                apis1.add(pathjob);
+
```

Comment:
APIDefinitionFromSwagger12 is used with swagger v1.2(In APIM 1.8.0) and removed since it is no longer in use(In APIM 1.9.0)
Instead we have APIDefinitionFromSwagger20 Implemented from same abstract class which implements APIDefinitionFromSwagger12